### DISCUSSION TITLE
How to simply single folder structure

### DISCUSSION THREAD

**Discussion by weilbith** [nvim-neo-tree/neo-tree.nvim#967] [Q&A]
Hey 👋🏾 
I struggle to name this "problem"/feature. But I know this from other IDEs etc, that in case of the following directory structure, it gets simplified/treated like a single folder.

```
- a
   - b
     - c
       - d
          - foo
          - bar
          - baz
```

it just becomes:

```
- a/b/c/d
  - foo
  - bar
  - baz
```

This is for example very annoying on Java/Kotlin projects where the file path represents the full package path including the global prefix of the root package. But opening it recursively is also not an option, since from folder `d` on (in this example), we have the full source/test file tree now.

I read through all configuration options and tried a repository search. But I struggle to find anything in this regards. So sorry if this is a duplicate. Thanks for your help! 🙏🏾

**Comment by cseickel** [comment]
It's called `group_empty_dirs`:
https://github.com/nvim-neo-tree/neo-tree.nvim/blob/8c89efb93b8383666a6f989762c2f610f5f2c930/README.md?plain=1#L278

**Reply by weilbith** [reply]
Thanks for the reply!
Hmm.That is the only option I found where I thought that would be it. But it didn't work for me. So I thought I misunderstood the naming because it states "empty". Maybe I should try again and fix it if you say this is 100% what I mean.

**Reply by cseickel** [reply]
True, I suppose its really grouping dirs that contain only one dir and no files. If it didn't work, maybe it was in the wrong location? It needs to be within the `filesystem` section.

**Reply by weilbith** [reply]
Sorry, I was on vacations for 3 weeks.
Thanks for the hint. I somehow put it in the wrong group.

But a little follow up question: 
If I know open such a folder, I still have to open all intermediate folders. So like I have to open the same line in neo-tree like 5 times until it reaches the level where multiple sub-folders are inside. Is it possible to toggle the folder state but "recursively" till the this level?

**Reply by cseickel** [reply]
I think it will do what you want if you add `scan_mode = "deep"` to the filesystem config, at the same level as `group_empty_dirs`.

https://github.com/zenoli/neo-tree.nvim/blob/0d40d82f868d6f0de24043fe66901d67e8dcd850/lua/neo-tree/defaults.lua#L369

**Reply by weilbith** [reply]
Phenomenal! Thank you so much for your help!
